### PR TITLE
update to dash 1.0, plus R build

### LINKED ISSUE #29: Boilerplate should include R component metadata
Once @rpkyle is done adding R code generation to `dash` (see https://github.com/plotly/dash/issues/445) this repo will need to be updated to:

1. Contain all the necessary directories required for the R classes to be put into the right place
2. Contain all the necessary boilerplate files that an R package needs such that it can be installed via `install_github()` from https://github.com/r-lib/devtools 
3. The NPM scripts and README are updated such that both the Python and R packages are generated simultaneously

Note: nothing in this issue implies that the boilerplate should depend on any part of the R ecosystem being installed on the system of the component developer.

### PR DESCRIPTION
**Pull request by alexcjohnson**
closes #29 
Updates tests here to use `dash.testing`, and more standardized build commands.

Adding R support turned out to be a lot simpler than #72 made it look - basically, adding in `r_prefix` and using it in the `dash-generate-components` call means it all *just works* ™️  - I added some tests that the right files (R and otherwise) are present after building.

This is ready for review, but should not be merged until https://github.com/plotly/dash/pull/801 is merged and released so we can point to a released version of dash in the various `requirements.txt` files.

### CODE REVIEW

**Comment by bcliang** [comment]
Please consider updating package.json with upgraded devDependency `eslint` to 5.16.0. The main thing here is that it closes a vulnerability in js-yaml@3.13.0
```
  "devDependencies": {
...
    "eslint": "^5.16.0",
...
  },
```

**Comment by alexcjohnson** [comment]
> Please consider updating package.json with upgraded devDependency `eslint` to 5.16.0

Now `eslint` is up to v6. Updated all of `package.json`, and the necessary `dash` changes have been released so we use that as well. @byronz ready for review.

File: tests/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,3 @@
-dash==0.38.0
+dash[testing]>=1.0.2
 pytest
```
Comment (byronz) [inline tests/requirements.txt]: no need to have pytest relevant dependencies here as `dash[testing]` will control that

File: {{cookiecutter.project_shortname}}/tests/requirements.txt [hunk 2]
```diff
@@ -3,10 +3,9 @@
 # pip install -r requirements.txt
 
 chromedriver-binary
-dash>=0.38.0
+dash[testing]>=1.0.2
 ipdb
 percy
```
Comment (byronz) [inline {{cookiecutter.project_shortname}}/tests/requirements.txt]: same here: `percy` adn `selenium` will be installed by `dash[testing]`

Comment (alexcjohnson) [reply {{cookiecutter.project_shortname}}/tests/requirements.txt]: For that matter I don't know why we need *anything* other than `dash[testing]`... none of our other repos require `chromedriver-binary`, and I don't see `ipdb`, `flake8`, or `pylint` used - do you see any reason we should keep those?

Comment (byronz) [reply {{cookiecutter.project_shortname}}/tests/requirements.txt]: if not needed, I'm fine to 🔪

**Pull request merged**